server/src/test/java/org/elasticsearch/rest/RestControllerTests.java

```diff
@@ -142,6 +143,9 @@ public MethodHandlers next() {
         assertEquals("true", threadContext.getHeader("header.1"));
         assertEquals("true", threadContext.getHeader("header.2"));
         assertNull(threadContext.getHeader("header.3"));
+        List<String> expectedProductResponseHeader = new ArrayList<>();
+        expectedProductResponseHeader.add("Elasticsearch");
+        assertEquals(expectedProductResponseHeader, threadContext.getResponseHeaders().getOrDefault("X-Elastic-Product", null));
```

Comment:
Maybe make the `ELASTIC_PRODUCT_HTTP_HEADER` variable package private so it can be referenced here instead of the string literal?